torchtitan/models/llama3/infra/parallelize.py

```diff
@@ -90,6 +88,17 @@ def parallelize_llama(
         )
         maybe_enable_async_tp(job_config, world_mesh["tp"])
 
+    if parallel_dims.cp_enabled:
+        for block in model.layers.values():
+            parallelize_module(
+                module=block.attention.sdpa.attention_fn_wrapper,
+                device_mesh=world_mesh["cp"],
+                parallelize_plan=_ContextParallel(
```

Comment:
So after this change, we only need to specify context parallel plan for attention module here, and CP of other modules is still handled by the context manager,

Reply:
You can check the discussion in https://github.com/pytorch/pytorch/pull/162542. It's definitely good to remove the context manager, but that may also have some implications to how users should write the model, like the wrapper in this PR.